src/main/java/com/zendesk/maxwell/schema/MysqlSavedSchema.java

```diff
@@ -721,17 +721,23 @@ private void fixColumnLength(Schema recaptured) throws SQLException {
 		}
 	}
 
-	protected void handleVersionUpgrades(Connection conn) throws SQLException, InvalidSchemaError {
+	protected void handleVersionUpgrades(ConnectionPool pool) throws SQLException, InvalidSchemaError {
 		if ( this.schemaVersion < 3 ) {
-			Schema recaptured = new SchemaCapturer(conn, sensitivity).capture();
+			final Schema recaptured;
+			try (Connection conn = pool.getConnection();
+				 SchemaCapturer sc = new SchemaCapturer(conn, sensitivity)) {
+				recaptured = sc.capture();
+			}
 
 			if ( this.schemaVersion < 1 ) {
 				if ( this.schema != null && this.schema.findDatabase("mysql") == null ) {
 					LOGGER.info("Could not find mysql db, adding it to schema");
-					SchemaCapturer sc = new SchemaCapturer(conn, sensitivity, "mysql");
-					Database db = sc.capture().findDatabase("mysql");
-					this.schema.addDatabase(db);
-					this.shouldSnapshotNextSchema = true;
+					try (Connection conn = pool.getConnection();
+						 SchemaCapturer sc = new SchemaCapturer(conn, sensitivity, "mysql")) {
```

Comment:
whitespace issue here

Reply:
not seeing it, can you be more specific?

Reply:
L736 doesn't line up with L737 and has additional space? it seems to happen in other `SchemaCapturer` as well